### ISSUE TITLE
Implement subdirectory sanitisation 

### ISSUE THREAD

**Issue by lukepayne** [RipMeApp/ripme#16]
The addURLToDownload method in AbstractRipper should have sanitisation implemented for sub directories. It's currently possible to pass invalid characters and have the directory creation fail. This is particularly an issue on Windows platforms.

**Comment by cyian-1756** [comment]
> Another side effect of no sanitisation is that slashes in the directory name create further subdirectories.

I don't think we should change that. There are a few rippers that rely on it working like that to download albums to sub folders.

**Comment by lukepayne** [comment]
> I don't think we should change that. There are a few rippers that rely on it working like that to download albums to sub folders.

Makes sense. I thought it was worth throwing that in there for investigation if needed.

**Comment by raviteja1992y** [comment]
how about we hash the name if it is not a valid file name.

**Comment by lukepayne** [comment]
@raviteja1992y Hashing would make the file name unreadable. I think the solution to this would be to filter out/replace illegal characters so that the file name is still decipherable.

**Comment by metaprime** [comment]
We currently have this kind of sanitation in place for top level directory and image file names, it seems like the same logic (or modified to preserve directory separators) for subdirectories would be a good idea.

**Comment by raviteja1992y** [comment]
As @metaprime, said there is Utils.filesystemSafe(..) method currently used in AlbumRipper.setWorkingDir(..) method and some other rippers for sanitisation.

**Comment by anuragashok** [comment]
I am open to help code on implementing this. Let me know if help is still needed ?

**Comment by anuragashok** [comment]
Ok. I will work on this.
As I understand from the discussion , the need is to sanitize the sub directory names ( except for the / char ).

**Comment by cyian-1756** [comment]
> the need is to sanitize the sub directory names ( except for the / char ).

Yup
